vm/inc/ubpf.h

```diff
@@ -106,7 +106,7 @@ int ubpf_load(struct ubpf_vm *vm, const void *code, uint32_t code_len, char **er
  */
 int ubpf_load_elf(struct ubpf_vm *vm, const void *elf, size_t elf_len, char **errmsg);
 
-int ubpf_exec(const struct ubpf_vm *vm, void *mem, size_t mem_len, uint64_t* bpf_return_value);
+int ubpf_exec(struct ubpf_vm *vm, void *mem, size_t mem_len, uint64_t* bpf_return_value);
```

Comment:
Hmm, I'd probably prefer to avoid this const change.

What if ubpf_inst_cnt_enable() just took a pointer to the ubpf_inst_cnt struct? It stores the pointer in the ubpf_vm. If the pointer in the ubpf_vm is non-null then instruction counting is enabled. Then we wouldn't have to worry about const issues (because we wouldn't be changing the ubpf_vm struct) and we wouldn't need a getter function, because the instruction counts are stored in the caller's memory....

So we should be able to accomplish this with only one new function added to the API.

Reply:
@lsgunth I did this in the next version.